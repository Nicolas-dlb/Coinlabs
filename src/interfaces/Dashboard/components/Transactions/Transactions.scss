@import "theme/_variables.scss";

@media only screen and (min-width: 992px) {
	.transactions {
		margin-top: 2vh;
		padding-top: 2vh;
		transition: background-color 0.3s ease;
		width: 340px;
		height: 42vh;
		background-color: var(--third-bg-color);
		border-radius: 20px;
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: flex-start;
		overflow: hidden;
	}
}
