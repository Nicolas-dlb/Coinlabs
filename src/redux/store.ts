import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
