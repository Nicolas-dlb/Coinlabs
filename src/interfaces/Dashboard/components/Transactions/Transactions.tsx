import "./Transactions.scss";
import { useSelector } from "react-redux";
import { selectWalletHistory } from "redux/reducers/walletsSlice";
import Transaction from "./Transaction/Transaction";

function Transactions() {
  const walletHistory = useSelector(selectWalletHistory);
  const history = walletHistory?.slice(0, 6);
  return (
    <div className="transactions">
      {history?.map((item: any) => (
        <Transaction
          name={item.name}
          status={item.status}
          price={Number(item.price)}
        />
      ))}
    </div>
  );
}

export default Transactions;
