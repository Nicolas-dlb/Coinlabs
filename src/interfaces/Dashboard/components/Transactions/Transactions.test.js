import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import store from "redux/store";
import { act } from "react-dom/test-utils";
import Transactions from "./Transactions";

describe("TransactionsItem", () => {
  let TransactionsComponent;
  beforeEach(async () => {
    await act(async () => {
      TransactionsComponent = await mount(
        <Provider store={store}>
          <Transactions />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(TransactionsComponent).toMatchSnapshot();
  });
});
