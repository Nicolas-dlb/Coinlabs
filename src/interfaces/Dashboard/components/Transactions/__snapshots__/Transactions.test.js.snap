// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsItem should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Transactions>
    <div
      className="transactions"
    />
  </Transactions>
</Provider>
`;
